Transient Test Failure - test_search_delete_hdca_output

Long standing issue also referenced in #20616.

```
__________________ TestJobsApi.test_search_delete_hdca_output __________________

self = <galaxy_test.api.test_jobs.TestJobsApi object at 0x7f71cc066b20>
history_id = '3fac04e22af259c6'

    @pytest.mark.require_new_history
    def test_search_delete_hdca_output(self, history_id):
        list_id_a = self.__history_with_ok_collection(collection_type="list", history_id=history_id)
        inputs = json.dumps(
            {
                "input1": {"src": "hdca", "id": list_id_a},
            }
        )
>       tool_response = self._job_search(tool_id="collection_creates_list", history_id=history_id, inputs=inputs)

history_id = '3fac04e22af259c6'
inputs     = '{"input1": {"src": "hdca", "id": "c34129969476e859"}}'
list_id_a  = 'c34129969476e859'
self       = <galaxy_test.api.test_jobs.TestJobsApi object at 0x7f71cc066b20>

lib/galaxy_test/api/test_jobs.py:862: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
lib/galaxy_test/api/test_jobs.py:1148: in _job_search
    self._search(search_payload, expected_search_count=1)
        empty_search_response = <Response [200]>
        history_id = '3fac04e22af259c6'
        inputs     = '{"input1": {"src": "hdca", "id": "c34129969476e859"}}'
        search_payload = {'history_id': '3fac04e22af259c6',
 'inputs': '{"input1": {"src": "hdca", "id": "c34129969476e859"}}',
 'state': 'ok',
 'tool_id': 'collection_creates_list'}
        self       = <galaxy_test.api.test_jobs.TestJobsApi object at 0x7f71cc066b20>
        tool_id    = 'collection_creates_list'
        tool_response = <Response [200]>
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <galaxy_test.api.test_jobs.TestJobsApi object at 0x7f71cc066b20>
payload = {'history_id': '3fac04e22af259c6', 'inputs': '{"input1": {"src": "hdca", "id": "c34129969476e859"}}', 'state': 'ok', 'tool_id': 'collection_creates_list'}
expected_search_count = 1

    def _search(self, payload, expected_search_count=1):
        # in case job and history aren't updated at exactly the same
        # time give time to wait
        for _ in range(5):
            search_count = self._search_count(payload)
            if search_count == expected_search_count:
                break
            time.sleep(1)
>       assert (
            search_count == expected_search_count
        ), f"expected to find {expected_search_count} jobs, got {search_count} jobs"
E       AssertionError: expected to find 1 jobs, got 0 jobs
E       assert 0 == 1

_          = 4
expected_search_count = 1
payload    = {'history_id': '3fac04e22af259c6',
 'inputs': '{"input1": {"src": "hdca", "id": "c34129969476e859"}}',
 'state': 'ok',
 'tool_id': 'collection_creates_list'}
search_count = 0
self       = <galaxy_test.api.test_jobs.TestJobsApi object at 0x7f71cc066b20>

lib/galaxy_test/api/test_jobs.py:1163: AssertionError
```

--- a/lib/galaxy/managers/jobs.py
+++ b/lib/galaxy/managers/jobs.py
@@ -29,6 +29,7 @@ from sqlalchemy import (
     or_,
     true,
 )
+from sqlalchemy.dialects.postgresql import aggregate_order_by
 from sqlalchemy.orm import aliased
 from sqlalchemy.sql import select
 from typing_extensions import TypedDict
@@ -798,7 +799,7 @@ class JobSearch:
         if self.dialect_name == "sqlite":
             return func.group_concat(column)
         else:
-            return func.array_agg(column, order_by=column)
+            return func.array_agg(aggregate_order_by(column, column.asc()))
 
     def _build_stmt_for_hdca(
         self, stmt, data_conditions, used_ids, k, v, user_id, value_index, require_name_match=True

